### ISSUE TITLE
Keyring is duplicated when new key is generated

### ISSUE THREAD

**Issue by ni4** [riboseinc/rnp#256]
## Description

When new key is generated it should be appended to the keyring. However actually all existing keys are appended as well.

## Steps to Reproduce
```
MacBook-Pro:rnp Ribose$ rnpkeys --list-keys
pgp_mem_readfile: can't open "/Users/Ribose/.rnp/pubring.gpg"
cannot read pubring /Users/Ribose/.rnp/pubring.gpg
cannot read pub keyring
fatal: failed to load keys

MacBook-Pro:rnp Ribose$ rnpkeys --generate-key
pgp_mem_readfile: can't open "/Users/Ribose/.rnp/pubring.gpg"
cannot read pubring /Users/Ribose/.rnp/pubring.gpg
cannot read pub keyring
Generating a new key...
signature  2048/RSA (Encrypt or Sign) 574a1b37ac7ba267 2017-06-30 
Key fingerprint: ed9f 605e fff0 59bc ae0a 2aff 574a 1b37 ac7b a267 
uid              RSA (Encrypt or Sign) 2048-bit key <Ribose@localhost>
rnp: generated keys in directory /Users/Ribose/.rnp
Enter passphrase for 574a1b37ac7ba267: 
Repeat passphrase for 574a1b37ac7ba267: 

MacBook-Pro:rnp Ribose$ rnpkeys --list-keys
1 key found
signature  2048/RSA (Encrypt or Sign) 574a1b37ac7ba267 2017-06-30
Key fingerprint: ed9f605efff059bcae0a2aff574a1b37ac7ba267
uid              RSA (Encrypt or Sign) 2048-bit key <Ribose@localhost> 

MacBook-Pro:rnp Ribose$ rnpkeys --generate-key --expert
Please select what kind of key you want:
	(1)  RSA (Encrypt or Sign)
	(19) ECDSA
	(22) EDDSA
> 19
Please select which elliptic curve you want:
	(1) NIST P-256
	(2) NIST P-384
	(3) NIST P-521
> 1
Generating a new key...
signature  384/ECDSA 8fd534d81ac53199 2017-06-30 
Key fingerprint: 643c 3a8a 9c99 0e87 26d6 76a3 8fd5 34d8 1ac5 3199 
uid              ECDSA 384-bit key <Ribose@localhost>
rnp: generated keys in directory /Users/Ribose/.rnp
Enter passphrase for 8fd534d81ac53199: 
Repeat passphrase for 8fd534d81ac53199: 
MacBook-Pro:rnp Ribose$ 

MacBook-Pro:rnp Ribose$ rnpkeys --list-keys
3 keys found
signature  2048/RSA (Encrypt or Sign) 574a1b37ac7ba267 2017-06-30
Key fingerprint: ed9f605efff059bcae0a2aff574a1b37ac7ba267
uid              RSA (Encrypt or Sign) 2048-bit key <Ribose@localhost> 

signature  2048/RSA (Encrypt or Sign) 574a1b37ac7ba267 2017-06-30
Key fingerprint: ed9f605efff059bcae0a2aff574a1b37ac7ba267
uid              RSA (Encrypt or Sign) 2048-bit key <Ribose@localhost> 

signature  384/ECDSA 8fd534d81ac53199 2017-06-30
Key fingerprint: 643c3a8a9c990e8726d676a38fd534d81ac53199
uid              ECDSA 384-bit key <Ribose@localhost> 

```
## Expected Behavior
We have two keys, first RSA and second ECDSA 256-bits
## Actual Behavior
First key (RSA) is duplicated, also ECDSA key has 384 bits instead of 256.

**Comment by ni4** [comment]
It would be more logical to fix this issue after refactoring is done.

**Comment by flowher** [comment]
I believe there are 2 different bugs here. Indeed keys are duplicated (it seems this is recent change, as I've never seen it before).
Second bug is "First key (RSA) is duplicated, also ECDSA key has 384 bits instead of 256.". This is now fixed by #257
